sklearn/linear_model/_quantile.py

```diff
@@ -0,0 +1,203 @@
+        and 'interior-point' with older versions.
+
+    solver_options : dict, optional
+        Additional parameters passed to scipy.optimize.linprog as options.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        Features got by optimizing the Huber loss.
+
+    intercept_ : float
+        Bias.
+
+    n_iter_ : int
+        Number of iterations that scipy.optimize.linprog has run for.
+
+    References
+    ----------
+    .. [1] Koenker, R., & Bassett Jr, G. (1978). Regression quantiles.
+            Econometrica: journal of the Econometric Society, 33-50.
+
+    .. [2] Chen, C., & Wei, Y. (2005).
+           Computational issues for quantile regression.
+           Sankhya: The Indian Journal of Statistics, 399-417.
+    """
+
+    def __init__(
+            self,
+            quantile=0.5,
+            alpha=0.0001,
+            fit_intercept=True,
+            copy_X=True,
+            solver='auto',
+            solver_options=None,
+    ):
+        self.quantile = quantile
+        self.alpha = alpha
+        self.fit_intercept = fit_intercept
+        self.copy_X = copy_X
+        self.solver = solver
+        self.solver_options = solver_options
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        sample_weight : array-like, shape (n_samples,)
+            Weight given to each sample.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        X, y = self._validate_data(X, y, accept_sparse=False,
+                                   y_numeric=True, multi_output=False)
+
+        sample_weight = _check_sample_weight(sample_weight, X)
+
+        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
+            X, y, fit_intercept=self.fit_intercept, copy=self.copy_X,
+            sample_weight=sample_weight)
+
+        if self.quantile >= 1.0 or self.quantile <= 0.0:
+            raise ValueError(
+                "Quantile should be strictly between 0.0 and 1.0, got %f"
+                % self.quantile)
+
+        n_obs, n_slopes = X.shape
+        n_params = n_slopes
+
+        X_full = X
+        if self.fit_intercept:
+            n_params += 1
+            X_full = np.concatenate([np.ones([n_obs, 1]), X], axis=1)
+
+        # the linear programming formulation of quantile regression
+        # follows https://stats.stackexchange.com/questions/384909/
+        c_vector = np.concatenate([
+            np.ones(n_params * 2) * self.alpha,
+            sample_weight * self.quantile,
+            sample_weight * (1 - self.quantile),
+        ])
```

Comment:
```suggestion
         alpha = np.sum(sample_weight) * self.alpha
         if self.fit_intercept:
            c = np.r_([
                0,
                np.full(2 * n_features, alpha),
                sample_weight * self.quantile,
                sample_weight * (1 - self.quantile),
            ])
            A_eq = np.concatenate([
            np.ones((n_samples, 1)),
            X,
            -X,
            np.eye(n_samples),
            -np.eye(n_samples),
        ], axis=1)
        else:
            c = np.r_([
                np.full(2 * n_features, alpha),
                sample_weight * self.quantile,
                sample_weight * (1 - self.quantile),
            ])
            A_eq = np.concatenate([
            X,
            -X,
            np.eye(n_samples),
            -np.eye(n_samples),
        ], axis=1)
```

Our objective function as given in the user guide is 1/n sum(loss) + alpha L1 which is equivalent to sum(loss) + n*alpha L1. Therefore the rescaling of alpha.